### PR TITLE
Jmprieur/add target462

### PR DESCRIPTION
**Pull request by jmprieur**
[Fixes](https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/commit/ac722d3e1c628346dd0181341e354d135cb871da) https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/issues/48 [and](https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/commit/ac722d3e1c628346dd0181341e354d135cb871da) https://github.com/AzureAD/microsoft-identity-abstractions-for-dotnet/issues/45 

- Adds netstandard2.1 (because HttpMethod.Patch is widely used).
- Adds the Patch methods to IDownstreamRestApi
- and net462 (to avoid drawing the entiere framework when this
library is consumed from a library targeting 462)

Dependencies:
```text
dotnet list src\Microsoft.Identity.Abstractions\Microsoft.Identity.Abstractions.csproj package --include-transitive
Project 'Microsoft.Identity.Abstractions' has the following package references
   [net462]:
   Top-level Package                  Requested   Resolved
   > Microsoft.SourceLink.GitHub      1.0.0       1.0.0
   > System.Net.Http                  4.3.4       4.3.4

   Transitive Package                                   Resolved
   > Microsoft.Build.Tasks.Git                          1.0.0
   > Microsoft.SourceLink.Common                        1.0.0
   > System.Security.Cryptography.Algorithms            4.3.0
   > System.Security.Cryptography.Encoding              4.3.0
   > System.Security.Cryptography.Primitives            4.3.0
   > System.Security.Cryptography.X509Certificates      4.3.0

   [netstandard2.0]:
   Top-level Package                     Requested   Resolved
   > Microsoft.SourceLink.GitHub         1.0.0       1.0.0
   > NETStandard.Library           (A)   [2.0.3, )   2.0.3

   Transitive Package                 Resolved
   > Microsoft.Build.Tasks.Git        1.0.0
   > Microsoft.NETCore.Platforms      1.1.0
   > Microsoft.SourceLink.Common      1.0.0

   [netstandard2.1]:
   Top-level Package                  Requested   Resolved
   > Microsoft.SourceLink.GitHub      1.0.0       1.0.0

   Transitive Package                 Resolved
   > Microsoft.Build.Tasks.Git        1.0.0
   > Microsoft.SourceLink.Common      1.0.0
```

### CODE REVIEW

File: src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs [hunk 1]
```diff
@@ -425,6 +425,148 @@ public partial interface IDownstreamRestApi
             Action<DownstreamRestApiOptionsReadOnlyHttpMethod>? downstreamRestApiOptionsOverride = null,
             CancellationToken cancellationToken = default)where TOutput : class;
 
+#if NETSTANDARD2_1_OR_GREATER
+
+        /// <summary>
+        /// Calls, using <see cref="HttpMethod.Patch"/>, a downstream REST API with some input data .
+        /// By default the input data is serialized in JSON  but you can provide your own serializer  in the action
```
Comment (jennyf19) [inline src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs:432]: >  but you can provide your own serializer  i [](http://example.com/codeflow?start=59&length=44)

nit: looks like there are a few extra spaces on line 432

File: src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs [hunk 2]
```diff
@@ -425,6 +425,148 @@ public partial interface IDownstreamRestApi
...
+        /// <param name="downstreamRestApiOptionsOverride">[Optional] Overrides the options proposed in the configuration described
+        /// by <paramref name="serviceName"/>.</param>
+        /// <param name="user">[Optional] Claims representing a user. This is useful in platforms like Blazor
+        /// or Azure Signal R, where the HttpContext is not available. In other platforms, the library
+        /// will find the user from the HttpContext.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns> The value returned by the downstream web API.</returns>
+        /// <example>
+        /// <code>
+        ///   await _downstreamWebApi.PatchForUserAsync&lt;MyItem&gt;(
```
Comment (jennyf19) [inline src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs:450]: _downstreamRestApi?

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs (hunk 1 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs:432]: Thanks fixed (8 occurences)

File: src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs (hunk 2 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Abstractions/DownstreamRestApi/IDownstreamRestApi.HttpMethods.cs:450]: Right!
Thanks

**Pull request merged**